add a webhook example

Euphyzr opened the pull request:
## Summary

Shows how to use `discord.Webhook` to send async and sync webhook messages and also using a discord client by fetching guild webhooks.

## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

apple502j commented:
1. We should have separate files for the three
2. and that's not how you use webhook.

Euphyzr commented:
I wished to make the webhook example self contained, thus one file. But I guess I can separate the client one. Regarding the usage, I didn't get how it's wrong. Can you please let me know? You wouldn't use webhooks to reply to a message true, but I wanted to highlight the mechanism.

numbermaniac commented:
Webhooks are often used for fetching something from an external website and posting it into a Discord channel; e.g. GitHub webhooks, Twitter webhooks, and so on. It's unusual for bots within Discord to be interacting with them in this way.

apple502j commented:
See https://discordpy.readthedocs.io/en/master/api.html#webhook-support for how to use webhook on discord.py 2.0.

Euphyzr commented:
Changed the example to delete and log when message contains "banned words" in a certain guild using webhook. Also, use webhook to log unhandled event exceptions. Hopefully, this is a more practical example. Updated the code to discord.py 2.0 as well.

Rapptz reviewed (changes requested):
As it stands I'm not really a big fan of this example. I think it tries to do too much rather than doing a simple thing.

Also please note that the library uses single quotes, not double quotes.

Tari-dev reviewed (changes requested):
Loop kwarg of aiohttp.ClientSession is deprecated since version 2.0.

JDJGInc commented:
You should probaly redesign it, so it's properly passing the session to the bot, like don't use a self.loop
Those don't exist in later aiohttp versions
